src/pages/Register.tsx

```diff
@@ -1,296 +1,318 @@
+  idx,
+  errors,
+  disabled,
+  defaultValue,
+  onChange,
+}: {
+  field: GetRegistrationFieldsResponse[0];
+  idx: number;
+  errors: FieldErrors<{
+    [fieldId: string]: string;
+  }>;
+  disabled: boolean;
+  defaultValue?: string;
+  onChange: (event: { target: { value: string } }) => void;
+}) {
+  switch (field.type) {
+    case 'TEXT':
+      return (
+        <TextInput
+          key={field.id}
+          idx={idx}
+          id={field.id}
+          name={field.name}
+          onChange={onChange}
+          defaultValue={defaultValue}
+          required={field.isRequired}
+          disabled={disabled}
+          errors={errors}
+        />
+      );
+    case 'SELECT':
+      return (
+        <SelectInput
+          key={field.id}
+          idx={idx}
+          id={field.id}
+          title={field.name}
+          name={field.name}
+          onChange={onChange}
+          defaultValue={defaultValue}
+          options={field.registrationFieldOptions}
+          required={field.isRequired}
+          disabled={disabled}
+          errors={errors}
+        />
+      );
+    default:
+      return null;
+  }
+}
+
+function TextInput(props: {
+  idx: number;
+  id: string;
+  name: string;
+  defaultValue?: string;
+  required: boolean | null;
+  disabled: boolean;
+  onChange: (event: { target: { value: string } }) => void;
+  errors: FieldErrors<{
+    [fieldId: string]: string;
+  }>;
+}) {
+  return (
+    <FlexColumn $gap="0.5rem">
+      <Label htmlFor={props.name} required={!!props.required}>
+        {props.name}
+      </Label>
+      <Input
+        defaultValue={props.defaultValue}
+        type="text"
+        name={props.name}
+        onChange={props.onChange}
+        disabled={props.disabled}
+      />
+      {props.errors?.[props.id] && <ErrorText>{props.errors?.[props.id]?.message}</ErrorText>}
+    </FlexColumn>
+  );
+}
+
+function SelectInput(props: {
+  idx: number;
+  id: string;
+  name: string;
+  title: string;
+  required: boolean | null;
+  disabled: boolean;
+  onChange: (event: { target: { value: string } }) => void;
+  defaultValue?: string;
+  options: RegistrationFieldOption[];
+  errors: FieldErrors<{
+    [fieldId: string]: string;
+  }>;
+}) {
+  return (
+    <FlexColumn $gap="0.5rem">
+      <Label htmlFor={props.name} required={!!props.required}>
+        {props.title}
+      </Label>
+      <Select
+        id={props.id}
+        name={props.name}
+        defaultValue={props.defaultValue}
+        onChange={props.onChange}
+        disabled={props.disabled}
+      >
+        <option value="" selected={props.defaultValue ? false : true} disabled>
+          Choose a value
+        </option>
+        {props.options.map((option) => (
+          <option key={option.id} value={option.value}>
+            {option.value}
+          </option>
+        ))}
+      </Select>
+      {props.errors?.[props.id] && <ErrorText>{props.errors?.[props.id]?.message}</ErrorText>}
+    </FlexColumn>
+  );
+}
```

Comment:
I think this would be better at /components/form

Reply:
maybe we can do this in another pr. the use case for this field is mainly for registration as it is seen in the errors type